### DISCUSSION TITLE
Turning off the debug CMD?

### DISCUSSION THREAD

**Discussion by masterofobzene** [CorsixTH/CorsixTH#2678] [General Diagnostics]
Is it there any way to stop the cmd window (windows here) from popping up when launching the game? or at least make it launch minimized?

**Comment by lewri** [comment]
MacOS and Linux afaik don't launch with a console by default. I'm not sure why Windows is different... @TheCycoONE ?

In Visual Studio for Windows we're targeting subsystem of console; I can change it in custom builds to use the Windows subsystem instead. Apparently there's some settings we can do too to turn off the cmd launch for regular builds.
![image](https://github.com/user-attachments/assets/167b75c9-6d37-4785-a7f3-9ba560f02557)

However, another issue is we also have crash information that references to check the console for information; however it seems a bit redundant if not all OSes do the same behaviour.

**Comment by lewri** [comment]
So my post here yesterday was a little half-baked, and Visual Studio was misbehaving yesterday.

In the CMakeLists for CorsixTH we don't set the target in Windows. This could be because MSVC can be Linux too? Not sure.

```cmake
# Declaration of the executable
if(APPLE)
  set(corsixth_icon_file ${CMAKE_SOURCE_DIR}/CorsixTH/Icon.icns)
  set_source_files_properties(
    ${corsixth_icon_file}
    PROPERTIES
    MACOSX_PACKAGE_LOCATION Resources
  )
  set(MACOSX_BUNDLE_ICON_FILE Icon.icns)


  add_executable(CorsixTH MACOSX_BUNDLE ${corsixth_icon_file})


  set_target_properties(CorsixTH PROPERTIES LINK_FLAGS_MINSIZEREL "-dead_strip")
  set_target_properties(CorsixTH PROPERTIES XCODE_ATTRIBUTE_LD_RUNPATH_SEARCH_PATHS "@executable_path/../Frameworks")
elseif(MSVC)
  add_executable(CorsixTH CorsixTH.rc)
else()
  add_executable(CorsixTH "")
endif()
```

If we do set the MSVC target though, the resulting executable won't have a console.
```diff
elseif(MSVC)
add_executable(CorsixTH WIN32 CorsixTH.rc)
```

In the project properties, this also sets the subsystem as mentioned previously to be `WINDOWS (/SUBSYSTEM:Windows)`

I'm not sure where the choice is to keep Windows builds explicitly targeting a console, however. Mac explicitly seems to say don't show one... and by default Linux won't either.

However, in Windows to get the console window back for debugging, you can then run this in Powershell
`.\CorsixTH.exe | echo`
Command Prompt isn't so nice when it's not a console application, you can only send output to a file:
`CorsixTH.exe > output.txt`

**Reply by ARGAMX** [reply]
btw
When I was going through the debug instructions
I spent some time trying to figure out how to run Corsix in the console in macOS.
I think we don't need to show the console by default.
But perhaps it will be necessary to add these instructions on how to launch the application in the console for Windows here:

https://github.com/CorsixTH/CorsixTH/wiki/Debugger-Tutorials#using-debuggerlua-for-debugging
